codegen/src/generator/json.rs

```diff
@@ -97,6 +97,20 @@ impl GenerateProtocol for JsonGenerator {
         "f64"
     }
 
+    fn generate_additional_annotations(&self, service: &Service, shape_name: &str, type_name: &str) -> Vec<String> {
+        let service_name = service.service_type_name();
+
+        if (service_name == "DynamoDb" || service_name == "DynamoDbStreams") &&
```

Comment:
Nested matchers end up being more verbose to do the same thing.  Thought about that and decided on the `if/else`.